lib/internal/url.js

```diff
@@ -1084,26 +1084,32 @@ function domainToUnicode(domain) {
   return _domainToUnicode(`${domain}`);
 }
 
-// Utility function that converts a URL object into an ordinary
-// options object as expected by the http.request and https.request
-// APIs.
+/**
+ * Utility function that converts a URL object into an ordinary options object
+ * as expected by the `http.request` and `https.request` APIs.
+ * @param {URL} url
+ * @returns {Record<string, unknown>}
+ */
 function urlToHttpOptions(url) {
+  const { hostname, pathname, port, username, password, search } = url;
   const options = {
+    __proto__: null,
+    ...url, // In case the url object was extended by the user.
     protocol: url.protocol,
```

Comment:
I believe we don't need any redeclarations like this line: `protocol: url.protocol` since `...url` already handles those.

Reply:
`...url` doesn't bring over properties that are defined by prototype getters/setters. (`({ ...new URL('about:blank') })` is an empty object)

Reply:
> `...url` doesn't bring over properties that are defined by prototype getters/setters

That's not it, it's because it only copies the enumerable own properties of the object (and `Object.hasOwnProperty(new URL('about:blank'), 'protocol') === false`). It does copy getters (e.g. `{ ...{ get key() { return 'value' } } }`).